### DISCUSSION TITLE
Unable to compile Noah mp on the WSL.(Ubuntu 22.04)

### DISCUSSION THREAD

**Discussion by xujinlong2000** [NCAR/hrldas#115] [Q&A]
Hi，
I failed to install the noahmp in wsl. I'm wondering if it's related to my compiler. I used gcc-9 and gcc-10, but both failed.
The installation of netcdf and jasper is fine, and the installation of WRF4.1 has been completed. I checked the online error message online and it's only since the release of v5 that this issue  has occurred(with mac). It's also possible that I didn't notice the previous error. Can someone help me?  I would appreciate it.
This is the error info. 
make[1]: *** No rule to make target '../../noahmp/drivers/hrldas/NoahmpIOVarType.o', needed by 'NoahmpUrbanDriverMainMod.o'. Stop.
make[1]: Leaving directory '/mnt/e/hrldas/urban/wrf'

I checked the location of NoahmpIOVarType. o and found that it was generated in../../noahmp/src/, while the dependency in the makefile in/mnt/e/hrldas/urban/wrf was found in/noahmp/drivers/hrldas for NoahmpIOVarType. o

This is my user_ Build_ Options
LIBJASPER = -ljpeg -L/mnt/e/Build_WRF/LIBRARIES/grib2/lib -ljasper
INCJASPER = -I/mnt/e/Build_WRF/LIBRARIES/grib2/include
NETCDFMOD = -I/mnt/e/Build_WRF/LIBRARIES/netcdf/include
NETCDFLIB = -L/mnt/e/Build_WRF/LIBRARIES/netcdf/lib -lnetcdf -lnetcdff

**Comment by cenlinhe** [comment]
This is weird. The "NoahmpIOVarType. o" should be generated under /noahmp/drivers/hrldas. Which version 5.0 did you use? Could you please try re-download the latest code in the GitHub?

**Reply by xujinlong2000** [reply]
Hello, I find it strange too that the people around me can succeed. I have tried many times but failed. I have used "git clone -- recurse submodules https://github.com/NCAR/hrldas"  and "git clone b release v5.0-- recurse submodules https://github.com/NCAR/hrldas" , but both failed. By the way, I am using Windows 11.

**Reply by cenlinhe** [reply]
I guess is that this is related to compiler. It seems that you are using gfortran compiler. Did you select the right ./configure option to generate the user_build_options file for gfortran? Could you please share your user_build_options file?

**Reply by xujinlong2000** [reply]
I select 5 when i configure. And here is my user_build_options.
[user_build_options.txt](https://github.com/NCAR/hrldas/files/13065732/user_build_options.txt)

**Reply by cenlinhe** [reply]
have you checked if the path to netcdf and jasper are both correctly setup in your computer? If these are correctly set, I am not sure why your issue occurred. My guess would be something related to your compiler but currently I do not have a solution. You may want to find the IT staff from your institution to help you. I am not a software engineer, and I have zero experience on the ubuntu/windows platform.
